ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageManager.java

```diff
@@ -108,6 +112,21 @@ public boolean controlCompactionForBlobStore(PartitionId id, boolean enabled) {
     throw new UnsupportedOperationException("Method not supported");
   }
 
+  @Override
+  public ClusterParticipant getPrimaryClusterParticipant() {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public boolean isFileExists(PartitionId partitionId, String fileName) {
+    throw new UnsupportedOperationException("Method not supported");
+  }
```

Comment:
This method is very generic for any implementation. Can we add a default implementation in the interface or make this method static in the default implementation itself.

Reply:
Like we had discussed, currently this is defined in `StoreManager` interface and the impl of this method in `StorageManager` further delegates this to `DiskManager.isFileExists()`

In fact, even `CloudStorageManager` can have an impl of this by checking the file in Azure blob storage.

The interface cannot have a default impl as we are delegating the impl to DiskManager.